flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobVertex.java

```diff
@@ -425,12 +426,12 @@ public void setStrictlyCoLocatedWith(JobVertex strictlyCoLocatedWith) {
                     "Strict co-location requires that both vertices are in the same slot sharing group.");
         }
 
-        CoLocationGroup thisGroup = this.coLocationGroup;
-        CoLocationGroup otherGroup = strictlyCoLocatedWith.coLocationGroup;
+        CoLocationGroupImpl thisGroup = this.coLocationGroup;
+        CoLocationGroupImpl otherGroup = strictlyCoLocatedWith.coLocationGroup;
```

Comment:
Wouldn't it be cleaner to add the `addVertex()` method to the `CoLocationGroup` interface?

Reply:
The idea was to have a readonly interface as the colocation groups do not need to be changed after initialization.

Reply:
I see, this is only the initialization.